common/src/main/java/com/wynntils/mc/mixin/CapeLayerMixin.java

```diff
@@ -1,22 +1,35 @@
 /*
- * Copyright © Wynntils 2023.
+ * Copyright © Wynntils 2023-2024.
  * This file is released under LGPLv3. See LICENSE for full license details.
  */
 package com.wynntils.mc.mixin;
 
+import com.llamalad7.mixinextras.sugar.Local;
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.wynntils.core.events.MixinHelper;
+import com.wynntils.mc.event.PlayerFeatureRenderTranslucentCheckEvent;
 import com.wynntils.mc.event.PlayerRenderLayerEvent;
+import com.wynntils.mc.extension.PlayerModelExtension;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.model.PlayerModel;
 import net.minecraft.client.player.AbstractClientPlayer;
 import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.RenderType;
 import net.minecraft.client.renderer.entity.layers.CapeLayer;
+import net.minecraft.client.renderer.entity.layers.RenderLayer;
+import net.minecraft.client.resources.PlayerSkin;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyArg;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 @Mixin(CapeLayer.class)
-public abstract class CapeLayerMixin {
+public abstract class CapeLayerMixin extends RenderLayer<AbstractClientPlayer, PlayerModel<AbstractClientPlayer>> {
```

Comment:
Please revert this change.

Reply:
This is actually not so evil as you think and is allowed in mixins. What's your reasoning? Or are the superclasses not used anywhere?

Reply:
The only reason to do this is to have access to more methods on `this`. We used to do this before, and it ended up getting messy, forcing us to add dummy constructors, etc, and in the end it did not work in all cases anyway. So I had to check how to resolve that. And then I found that a much better way is to make a double cast:

`(VanillaTargetClass) (Object) this`. It is a bit more verbose at the call point, but we get rid of all the other cruft. Steve has fixed it this way already in the PR now.

Reply:
So I went a pass through all our mixins to make sure we did not do that anywhere, but used double-casting instead. I'd like to keep it that way. So the rule is: all mixins should be `public abstract class <VanillaClassName>Mixin {` and that's it. Ideally, I'd like to have some check to enforce this.